src/Core/BackgroundSchedulePool.cpp

```diff
@@ -243,7 +244,14 @@ void BackgroundSchedulePool::attachToThreadGroup()
 
 void BackgroundSchedulePool::threadFunction()
 {
-    setThreadName(thread_name.c_str());
+    try
+    {
+        setThreadName(thread_name.c_str());
+    }
+    catch (const DB::Exception &)
```

Comment:
This code does nothing.

Reply:
Print log, and throw exception to caller ?

Reply:
Exception will appear in log nevertheless.

Reply:
Ok, this code should be removed.